lib/bolt12-info.js

```diff
@@ -0,0 +1,33 @@
+import { deserializeTLVStream } from './tlv'
+import * as bech32b12 from '@/lib/bech32b12'
+
+const TYPE_DESCRIPTION = 10n
+const TYPE_PAYER_NOTE = 89n
+const TYPE_PAYMENT_HASH = 168n
+
+export function isBolt12 (invoice) {
+  return invoice.startsWith('lni1') || invoice.startsWith('lno1')
+}
+
+export function bolt12Info (bolt12) {
```

Comment:
This is the bolt12 equivalent of the bolt11Tags helper.
Similar to the bolt11 counterpart, i've used a light parser just for the few tags we need in the ui, since the lndk parser needs access to the grpc endpoint.

I couldn't find an established pure js bolt12 parser, so i wrote a small tlv parser (`@/lib/tlv`) following the specs.